google/auth/credentials.py

```diff
@@ -127,6 +130,8 @@ def apply(self, headers, token=None):
         headers["authorization"] = "Bearer {}".format(
             _helpers.from_bytes(token or self.token)
         )
+        if self._trust_boundary is not None:
+            headers["X-Identity-Trust-Boundary"] = self._trust_boundary
```

Comment:
nit: should this all be lowercase? Probably doesn't matter but normally these x-xyz header keys seem to be lowercase in this lib

Reply:
> nit: should this all be lowercase? Probably doesn't matter but normally these x-xyz header keys seem to be lowercase in this lib

I asked Annan for confirmation. She explicitly mentioned dashed camel case. IDK whether that matters for her. Maybe should ask whether can I change that to meet our consistency.

Reply:
That's fine if its what the service expects, just wanted to double check